File: JAVA/1417. Reformat The String.java
class Solution {     
    public String reformat(String s) {         
        StringBuilder a = new StringBuilder(), b = new StringBuilder(), c = new StringBuilder();         
        for (int i = 0; i < s.length(); i++) {             
            if (s.charAt(i) <= '9' && s.charAt(i) >= '0')                 
                a.append(s.charAt(i));             
            else                 
                b.append(s.charAt(i));         
        }         
        if (Math.abs(a.length() - b.length()) > 1)             
            return "";         
        if (a.length() == b.length()) {             
            for (int i = 0; i < a.length(); i++) {                 
                c.append(b.charAt(i));                 
                c.append(a.charAt(i));             
            }         
        } else if (a.length() > b.length()) {             
            for (int i = 0; i < b.length(); i++) {                 
                c.append(a.charAt(i));                 
                c.append(b.charAt(i));             
            }             
            c.append(a.charAt(a.length() - 1));         
        } else {             
            for (int i = 0; i < a.length(); i++) {                 
                c.append(b.charAt(i));                 
                c.append(a.charAt(i));             
            }             
            c.append(b.charAt(b.length() - 1));         
        }         
        return c.toString();     
    } 
}

File: C++/1417. Reformat The String.cpp
class Solution {
public:
    string reformat(string s) {
        string a, b, c;
        for (char ch : s) {
            if (ch >= '0' && ch <= '9') a += ch;
            else b += ch;
        }
        if (abs((int)a.size() - (int)b.size()) > 1) 
            return "";
        if (a.size() == b.size()) {
            for (int i = 0; i < a.size(); i++) {
                c.push_back(b[i]);
                c.push_back(a[i]);
            }
        }
        else if (a.size() > b.size()) {
            for (int i = 0; i < b.size(); i++) {
                c.push_back(a[i]);
                c.push_back(b[i]);
            }
            c.push_back(a.back());
        }
        else {
            for (int i = 0; i < a.size(); i++) {
                c.push_back(b[i]);
                c.push_back(a[i]);
            }
            c.push_back(b.back());
        }
        return c;
    }
};
